### PR TITLE
Add configurable EZID action for minting ARKs from content page (Issue 28)

### PR DESCRIPTION
**Pull request by seth-shaw-unlv**
Add configurable EZID action for minting ARKs from content page (#28)

Also includes EZID Service Data Type form update to clarify namespace configuration (#27).

### CODE REVIEW

**Comment by seth-shaw-unlv** [comment]
Now also addresses #29

File: src/Plugin/Action/MintIdentifier.php [hunk 1]
```diff
@@ -27,7 +29,11 @@ protected function getFieldData(): array {
     if ($data_profile) {
       foreach ($data_profile->getData() as $key => $field) {
         if ($this->entity->hasField($field)) {
-          $data[$key] = $this->entity->get($field)->getString();
+          if ($this->entity->get($field) instanceof EntityReferenceFieldItemList) {
+            $data[$key] = $this->entity->get($field)->entity->label();
+          } else {
+            $data[$key] = $this->entity->get($field)->getString();
+          }
```
Comment (jordandukart) [inline src/Plugin/Action/MintIdentifier.php:36]:
```suggestion
          $entity_field = $this->entity->get($field);
          if (!$entity_field->isEmpty()) {
            if ($this->entity->get($field) instanceof EntityReferenceFieldItemList) {
              $data[$key] = $entity_field->entity->label();
            }
            else {
              $data[$key] = $entity_field->getString();
            }
          }
```

So nit about coding standards but noticed another bug while testing this. A non-empty field existence isn't checked for at the moment 🤦.

**Review by jordandukart — CHANGES_REQUESTED** [review]
Thanks for the contribution @seth-shaw-unlv, great improvements/bug fixes overall!

File: src/Plugin/Action/MintIdentifier.php [hunk 2]
```diff
@@ -72,6 +78,9 @@ public function execute($entity = NULL): void {
         $this->entity = $entity;
         if ($this->entity && $this->identifier) {
           $this->setIdentifierField($this->mint());
+          if (array_key_exists('save_entity', $this->configuration) && $this->configuration['save_entity']) {
```
[the remark this answers is not in the document]
Comment (seth-shaw-unlv) [reply src/Plugin/Action/MintIdentifier.php:81]: Indeed. I didn't want existing actions to choke if I didn't have an update hook.

File: src/Plugin/Action/MintIdentifier.php (hunk 1 above)
Comment (seth-shaw-unlv) [reply src/Plugin/Action/MintIdentifier.php]: I took a slightly different approach on this one, but the effect is the same.

**Comment by seth-shaw-unlv** [comment]
@jordandukart, ready for another look.

**Pull request merged**